RPG basic weighted inventory system

Kavignon opened the pull request:
The inventory system allows the the team's party to access game items that can be equipped during battle. 
Those items include consummables such medicinal herb to restore health points or phoenix feathers that can brought back the dead with 50% of health and mana. 
The inventory system will also keep in check the equipment usage and the weapons hit limits and once the usage gets to 0% and the hit limit gets to 0, then either the equipment or the weapon will disappear both from the inventory and the equipment.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gameoflightandshadows/a-game-of-light-and-shadows/101)
<!-- Reviewable:end -->

Commit:
#27 -  Defining all the type of different items that can be found for game items
Defined the function to add and drop items in the inventory

review-ninja commented:
<a href="https://app.review.ninja/GameOfLightAndShadows/A-game-of-light-and-shadows/pull/101" target="_blank"><img src="https://app.review.ninja/66727011/pull/101/badge" alt="ReviewNinja"/>

Commit:
#27 - Defining consummables and character wearables

Defined properties for consummables and daggers

Commit:
#27 - Consummable items

Consummable items now impact the stats of a character once they have
been selected for usage.
Also, changes were made for the Swords

Commit:
#27 - Finish defining weapon

Finished defining the properties for the weapons such as their weight,
their price and their stats

Commit:
#102 - Completed the designing and implementing a sufficient and realistic domain model for the consummable items and the weaponry
Units of measure were added to make the the type system strict

Signed-off-by: Kevin Avignon <kevin.o.avignon@gmail.com>

Commit:
#102 - Finished design and implement the domain model for the character’s protection

Signed-off-by: Kevin Avignon <kevin.o.avignon@gmail.com>

Commit:
#100 - Completed implementing the inventory system. Items can added by one or many. Item can be remove one at a time. When the player gets too many items, they will be moved to the Excess stack. Items will be moved gradually when the bag becomes lighter

Signed-off-by: Kevin Avignon <kevin.o.avignon@gmail.com>

Commit:
#90 - Implemented an inventory manager which will receives updates from either the team manager or the shop manager to make update to the inventory. Once the update is completed,  it’s sent to the game manager who’s job will be to make sure that every character knows about it.
